libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -672,6 +689,28 @@ internal object MapboxNavigationTelemetry {
         sendEvent(freeDriveEvent)
     }
 
+    @ExperimentalPreviewMapboxNavigationAPI
+    private fun createCustomEvent(
+        payload: String,
+        customEventType: String,
+        customEventVersion: String,
+        phoneState: PhoneState,
+        onEventUpdated: ((NavigationCustomEvent) -> Unit)?
+    ) {
+        log("customEventType: $customEventType")
+        val customEvent =
+            NavigationCustomEvent().apply {
+                this.payload = payload.plus(", userId = ${phoneState.userId}")
```

Comment:
Is payload not a json string? Is it easy to parse the result?

Reply:
This is a good question. There is no requirement from the schema side that it should be a json string. The only requirement is that it should be a string. But, I think sending it as a json string will make it easier to follow guidelines on how to add more data to it rather than separating the data using separators like `,`